sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/DistributionSuite.scala

```diff
@@ -41,34 +41,127 @@ class DistributionSuite extends SparkFunSuite {
     }
   }
 
-  test("HashPartitioning (with nullSafe = true) is the output partitioning") {
-    // Cases which do not need an exchange between two data properties.
+  test("UnspecifiedDistribution and AllTuples") {
```

Comment:
I've reorganized this test suite and added a bunch of new test cases, to improve the test coverage.